conversions/octal_to_hexadecimal.py

```diff
@@ -0,0 +1,65 @@
+def octal_to_hex(octal: str) -> str:
+    """
+    Convert an Octal number to Hexadecimal number.
+    For more information: https://en.wikipedia.org/wiki/Octal
+
+    >>> octal_to_hex("100")
+    '0X40'
+    >>> octal_to_hex("235")
+    '0X9D'
+    >>> octal_to_hex(17)
+    Traceback (most recent call last):
+        ...
+    TypeError: Expected a string as input
+    >>> octal_to_hex("Av")
+    Traceback (most recent call last):
+        ...
+    ValueError: Not a Valid Octal Number
+    >>> octal_to_hex("")
+    Traceback (most recent call last):
+        ...
+    ValueError: Empty string was passed to the function
+    """
+
+    if not isinstance(octal, str):
+        raise TypeError("Expected a string as input")
+    if octal.startswith("0o"):
+        octal = octal[2:]
+    if octal == "":
+        raise ValueError("Empty string was passed to the function")
+    if any(char not in "01234567" for char in octal):
+        raise ValueError("Not a Valid Octal Number")
+
+    decimal = 0
+    for char in octal:
+        decimal <<= 3
+        decimal |= int(char)
+
+    hex_char = "0123456789ABCDEF"
+
+    revhex = ""
+    while decimal:
+        revhex += hex_char[decimal & 15]
+        decimal >>= 4
+
+    return "0X" + revhex[::-1]
```

Comment:
```suggestion
    return "0x" + revhex[::-1]
```
`0x` (lowercase "x") is the standard prefix for hex numbers, I believe. Could you also make this change in the doctests?

Reply:
Yeah sure